server/api/db/sqldb/db.py

```diff
@@ -2925,6 +2926,21 @@ def _update_feature_set_features(
                 )
                 update_labels(feature, labels)
                 feature_set.features.append(feature)
+            elif feature_name not in features_to_remove:
+                # check if the existing feature's labels were changed, and update them if needed
+                feature = next(
+                    (
+                        feature
+                        for feature in feature_set.features
+                        if feature.name == feature_name
+                    ),
+                    None,
+                )
+                labels = feature_dict.get("labels") or {}
+                if not server.api.utils.helpers.are_object_labels_equal_to_labels_dict(
```

Comment:
Use `deepdiff`, no need to implement it yourself

Reply:
removed the check altogether